# HG changeset patch
# User Kurt Smith <kwmsmith@gmail.com>
# Date 1280413165 18000
# Node ID e149b7b35a78d5135fc85bf9549150ce1a915f69
# Parent  b6758348c05825e7e7f8f4be0ccec073fd74421b
use 'hg identify' to generate .hgrev file

diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -6,10 +6,9 @@
 if 'sdist' in sys.argv and sys.platform != "win32" and sys.version_info >= (2,4):
     # Record the current revision in .hgrev
     import subprocess # os.popen is cleaner but deprecated
-    changset = subprocess.Popen("hg log --rev tip | grep changeset", 
-                                shell=True,
-                                stdout=subprocess.PIPE).stdout.read()
-    rev = changset.decode('ISO-8859-1').split(':')[-1].strip()
+    changeset = subprocess.Popen("hg identify --id --rev tip".split(),
+                                 stdout=subprocess.PIPE).stdout.read()
+    rev = changeset.decode('ISO-8859-1').strip()
     hgrev = open('.hgrev', 'w')
     hgrev.write(rev)
     hgrev.close()
